harvest/plugin/_base.py

```diff
@@ -1,9 +1,39 @@
+from typing import List
+from logging import error
+
 class Plugin:
     """
     As the name implies, it implements funtionalities that the Trader class
     alone cannot support, like retrieving earning report dates.  
 
-    The name of the plugin will follow thge format:
+    The name of the plugin will follow the format:
         {Name of source}{Functionality added}Plugin
     """
-    pass 
\ No newline at end of file
+    def __init__(self, name: str, dependencies: List[str]=[]):
+        """
+        :name: a string that will serve as the name of the plugin. Must
+        be a valid python variable because in the algo class, this plugin
+        will be an attribute of the algo and accessable via algo.name.
+        :dependencies: a list of python packages as strings that will be 
+        checked to ensure they are installed.
+        Put all imports here in order to test if we have them. If not then
+        error and call the installation function.
+        """
+        self.name = name
+
+        for dep in dependencies:
+            self._check_dependency(dep)
+
+    def installation(self) -> str:
```

Comment:
Since installation for a plugin can me very complicated, I think it is best just to return a string of instructions on how to get the prerequisites installed for a plugin.

Reply:
I agree, additionally maybe we can include a README for each plugin with instructions and documentations.